src/passes/Metrics.cpp

```diff
@@ -92,6 +93,34 @@ struct Metrics : public WalkerPass<PostWalker<Metrics, UnifiedExpressionVisitor<
         counts["[binary-bytes]"] = writer.tableOfContents.functionBodies[i].size;
         printCounts(std::string("func: ") + func->name.str);
       }
+      // print for each export how much code size is due to it, i.e.,
+      // how much the module could shrink without it.
+      auto sizeAfterGlobalCleanup = [](Module* module) {
+        PassRunner runner(module, PassOptions::getWithDefaultOptimizationOptions());
+        runner.setIsNested(true);
+        runner.addDefaultGlobalOptimizationPostPasses(); // remove stuff
+        runner.run();
+        BufferWithRandomAccess buffer;
+        WasmBinaryWriter writer(module, buffer);
+        writer.write();
+        return buffer.size();
+      };
+      size_t baseline;
+      {
+        Module test;
+        ModuleUtils::copyModule(*module, test);
+        baseline = sizeAfterGlobalCleanup(&test);
+      }
+      for (auto& exp : module->exports) {
+        // create a test module where we remove the export and then see how much can be removed thanks to that
+        Module test;
+        ModuleUtils::copyModule(*module, test);
+        test.removeExport(exp->name);
```

Comment:
Ok, so this is removing the export, generating the wasm again for this parallel universe, and then getting the resulting size? SGTM

Reply:
Does this handle the case when one exported function calls another exported function?

Reply:
Yes, exactly. It removes and optimizes.

Yes, calling an exported function is handled - the export itself isn't called, it's the function, so whether the export is removed or not is separate from whether we can remove the function.